### PR TITLE
Refactors profile screen component naming and fixes types missing

### LINKED ISSUE #460: Refactor profile screen and its sub components
### Issue Description

The profile screen needs to be refactored to improve code readability, optimize performance, and adhere to the latest coding standards.

### Expected Behavior

- The refactored code should be modular, maintainable, and adhere to the existing coding standards.
- Profile screen functionality and UI should remain unchanged after the refactor.
- Code should be optimized to reduce any unnecessary re-renders or performance bottlenecks.
- fix types wherever required.

### Current Behavior

- Code in the profile screen is difficult to read and maintain.
- Component state management is not optimal, leading to potential performance issues.
- Code style and structure do not align with the latest project conventions.
- types incorrectly added

### Environment

- **Operating System:** [e.g., Windows, macOS, Linux]
- **Browser (if applicable):** [e.g., Chrome, Firefox, Safari]
- **Version of Software/Project:** [e.g., v1.0.0]

### Screenshots

image

### Checklist

- [x] I have read and followed the project's code of conduct.
- [x] I have searched for similar refactor requests before creating this one.
- [x] I have provided all the necessary information to understand and perform the refactor.
- [x] I am willing to contribute to the resolution of this issue.

---

Thank you for contributing to our project! We appreciate your help in improving it.

📚 See [contributing instructions](https://github.com/owner/repo/blob/develop/CONTRIBUTING.md) for more details.

### PR DESCRIPTION
**Pull request by Ayushsanjdev**
Date: 25-10-2024

Developer Name: Ayush Sanj

---

## Issue Ticket Number

Closes #460 

## Description

1. Refactors profile component screen.
2. Fix types
3. remove commented code.

### Documentation Updated?

- [ ] Yes
- [x] No

### Under Feature Flag

- [ ] Yes
- [x] No

### Database Changes

- [ ] Yes
- [x] No

### Breaking Changes

- [ ] Yes
- [x] No

### Development Tested?

- [ ] Yes
- [x] No

## Screenshots
<details>
<summary>Screenshot 1</summary>

</details>

## Test Coverage
<details>
<summary>Screenshot 1</summary>

</details>

## screen recording

https://github.com/user-attachments/assets/b6f9cc94-9ac0-49b8-a4be-424694c08b3f

### CODE REVIEW

File: src/screens/ProfileScreen/DetailsScreen/ProgressDetailScreen.tsx [hunk 1]
```diff
@@ -10,22 +10,14 @@ import React, { useState } from 'react';
...
   const [taskCompleted, setTaskCompleted] = useState('');
   const [taskPlanned, setTaskPlanned] = useState('');
   const [taskBlockers, setTaskBlockers] = useState('');
-  // const [shouldDisable, setShoulDisable] = useState(false);
 
-  const isVisible = () => {
-    // setShoulDisable('');
-    // validateTaskProgress(taskCompleted, taskPlanned, taskBlockers),
-  };
+  const isVisible = () => {};
```
Comment (pankajjs) [inline src/screens/ProfileScreen/DetailsScreen/ProgressDetailScreen.tsx]: Why do we have this?

File: src/screens/ProfileScreen/DetailsScreen/ProgressDetailScreen.tsx [hunk 2]
```diff
@@ -10,22 +10,14 @@ import React, { useState } from 'react';
...
 
-  function submitProgress(): void {
-    // throw new Error('Function not implemented.');
-  }
+  function submitProgress(): void {}
```
Comment (pankajjs) [inline src/screens/ProfileScreen/DetailsScreen/ProgressDetailScreen.tsx]: same

File: src/screens/ProfileScreen/DetailsScreen/ProgressDetailScreen.tsx (hunk 1 above)
Comment (Ayushsanjdev) [reply src/screens/ProfileScreen/DetailsScreen/ProgressDetailScreen.tsx]: I am assuming if we require this function later to write logic inside it.

File: src/screens/ProfileScreen/DetailsScreen/ProgressDetailScreen.tsx (hunk 2 above)
Comment (Ayushsanjdev) [reply src/screens/ProfileScreen/DetailsScreen/ProgressDetailScreen.tsx]: I am assuming if we require this function later to write logic inside it.

File: src/screens/ProfileScreen/DetailsScreen/ProgressDetailScreen.tsx (hunk 1 above)
Comment (pankajjs) [reply src/screens/ProfileScreen/DetailsScreen/ProgressDetailScreen.tsx]: We can write it whenever it is required. No point including it in this PR when it is not being used.

File: src/screens/ProfileScreen/DetailScreen/ActiveTaskDetail.tsx [hunk 3]
```diff
@@ -3,64 +3,41 @@ import { StyleSheet } from 'react-native';
 import { View, Text, TouchableOpacity, ScrollView } from 'react-native';
 import { useNavigation, useRoute } from '@react-navigation/native';
 import ProgressModal from '../../../components/Modal/ProgressModal';
-// import { getDateAndTimeFromUnix } from '../../AuthScreen/Util';
 
 const ActiveTaskDetail = () => {
   const route = useRoute();
-  const { task } = route.params;
+  const { task } = route.params as any; // TODO fix types when used via route navigation
```
Comment (pankajjs) [inline src/screens/ProfileScreen/DetailScreen/ActiveTaskDetail.tsx:10]: Can't we define a type for this?

File: src/screens/ProfileScreen/DetailScreen/ActiveTaskDetail.tsx [hunk 4]
```diff
@@ -3,64 +3,41 @@ import { StyleSheet } from 'react-native';
...
-  console.log(task, 'percentCompleted');
-
-  const formatStatusText = (status) => {
-    // Split the status string by underscore
+  const formatStatusText = (status: string) => {
     const words = status.split('_');
-
-    // Capitalize the first letter of each word
     const formattedText = words.map(
       (word) => word.charAt(0).toUpperCase() + word.slice(1).toLowerCase(),
```
Comment (pankajjs) [inline src/screens/ProfileScreen/DetailScreen/ActiveTaskDetail.tsx]: We can put it into a separate util function.

File: src/screens/ProfileScreen/DetailScreen/ExtensionRequest.tsx [hunk 5]
```diff
@@ -30,8 +29,7 @@ export default function ExtensionRequest() {
       .then((responseData) => {
         console.log(JSON.stringify(responseData));
         console.log(JSON.stringify(responseData), 'PostApiCAlled');
-      })
-      .done();
```
Comment (pankajjs) [inline src/screens/ProfileScreen/DetailScreen/ExtensionRequest.tsx:34]: Why did we remove .done()?

File: src/screens/ProfileScreen/DetailScreen/ExtensionRequest.tsx [hunk 6]
```diff
@@ -94,7 +91,9 @@ const styles = StyleSheet.create({
   container: {
     flex: 1,
     backgroundColor: 'white',
-    border: '2px solid black',
```
Comment (pankajjs) [inline src/screens/ProfileScreen/DetailScreen/ExtensionRequest.tsx:97]: Why did we change into separate property?

File: src/screens/ProfileScreen/TaskScreens/All.tsx [hunk 7]
```diff
@@ -10,31 +10,22 @@ const All = () => {
...
-
   useFocusEffect(
     useCallback(() => {
       setLoading(true);
 
       (async () => {
         const token = loggedInUserData?.token;
 
-        const allTasks = await fetchAllTasks(token);
+        const allTasks = await fetchAllTasks(token as string);
```
Comment (pankajjs) [inline src/screens/ProfileScreen/TaskScreens/All.tsx:20]: We can define type in line 18.

File: src/screens/ProfileScreen/TaskScreens/NoteWorthy.tsx [hunk 8]
```diff
@@ -24,61 +24,39 @@ const Note = () => {
     useCallback(() => {
       (async () => {
         const userName = loggedInUserData?.username;
-        const contributionResponse = await fetchContribution(userName);
+        const contributionResponse = await fetchContribution(
+          userName as string,
```
Comment (pankajjs) [inline src/screens/ProfileScreen/TaskScreens/NoteWorthy.tsx:39]: same

File: src/screens/ProfileScreen/TaskScreens/NoteWorthy.tsx [hunk 9]
```diff
@@ -103,19 +75,69 @@ const Note = () => {
...
+          <Text style={styles.noTaskText}>No noteworthy task yet!</Text>
         </View>
       )}
     </ScrollView>
   );
 };
 
 export default Note;
+
+const styles = StyleSheet.create({
```
Comment (pankajjs) [inline src/screens/ProfileScreen/TaskScreens/NoteWorthy.tsx:99]: We can put this into a separate file.

File: src/screens/ProfileScreen/DetailScreen/ActiveTaskDetail.tsx (hunk 3 above)
Comment (Ayushsanjdev) [reply src/screens/ProfileScreen/DetailScreen/ActiveTaskDetail.tsx:10]: No as it is not being used via route navigation, so no data being come in task property, so when someone uses this, they can define the types at that time. That's why I wrote TODO and put it as any for now

File: src/screens/ProfileScreen/DetailScreen/ExtensionRequest.tsx (hunk 5 above)
Comment (Ayushsanjdev) [reply src/screens/ProfileScreen/DetailScreen/ExtensionRequest.tsx:34]: Checkout fetch call methods, there is no done property.
https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API

File: src/screens/ProfileScreen/DetailScreen/ExtensionRequest.tsx (hunk 6 above)
Comment (Ayushsanjdev) [reply src/screens/ProfileScreen/DetailScreen/ExtensionRequest.tsx:97]: You can google about this.

File: src/screens/ProfileScreen/TaskScreens/NoteWorthy.tsx (hunk 9 above)
Comment (Ayushsanjdev) [reply src/screens/ProfileScreen/TaskScreens/NoteWorthy.tsx:99]: We can only put it in separate file, if styles are in very large no of lines.

Comment (pankajjs) [reply src/screens/ProfileScreen/TaskScreens/NoteWorthy.tsx:99]: There is someone who is already working on similar issue where the stylesheet is being removed from component file. So, just for consistency, I asked it.

Comment (Ayushsanjdev) [reply src/screens/ProfileScreen/TaskScreens/NoteWorthy.tsx:99]: Yes but only for the files only whose stylesheet is longer causing component unreadable.

File: src/screens/ProfileScreen/TaskScreens/All.tsx (hunk 7 above)
Comment (Ayushsanjdev) [reply src/screens/ProfileScreen/TaskScreens/All.tsx:20]: right but it depends on reusability

**Review by pankajjs — APPROVED** [review]

**Comment by Ayushsanjdev** [comment]
@ashifkhn Added screen recording

**Review by ashifkhn — APPROVED** [review]

File: src/screens/ProfileScreen/DetailScreen/ActiveTaskDetail.tsx [hunk 10]
```diff
@@ -3,64 +3,34 @@ import { StyleSheet } from 'react-native';
...
         <Text style={styles.titles} />
 
-        <TouchableOpacity
-          style={styles.buttonStyle}
-          onPress={() =>
-            // navigation.navigate('ExtensionRequest', { endsOn: task.endsOn })
-            console.log('here')
-          }
-        >
+        <TouchableOpacity style={styles.buttonStyle}>
```
Comment (ashifkhn) [inline src/screens/ProfileScreen/DetailScreen/ActiveTaskDetail.tsx]: get used to pressable :)

File: src/screens/ProfileScreen/DetailScreen/ActiveTaskDetail.tsx [hunk 11]
```diff
@@ -119,6 +89,9 @@ const styles = StyleSheet.create({
     justifyContent: 'center',
     borderRadius: 15,
   },
+  backButton: {
+    color: 'white',
```
Comment (ashifkhn) [inline src/screens/ProfileScreen/DetailScreen/ActiveTaskDetail.tsx:93]: use color variable

File: src/screens/ProfileScreen/DetailScreen/ExtensionRequest.tsx [hunk 12]
```diff
@@ -94,7 +91,9 @@ const styles = StyleSheet.create({
   container: {
     flex: 1,
     backgroundColor: 'white',
```
Comment (ashifkhn) [inline src/screens/ProfileScreen/DetailScreen/ExtensionRequest.tsx:93]: use color variable

File: src/screens/ProfileScreen/DetailScreen/ExtensionRequest.tsx [hunk 13]
```diff
@@ -144,8 +143,11 @@ const styles = StyleSheet.create({
   },
   input: {
     borderColor: 'skyBlue',
-    borderwidth: 2,
+    borderWidth: 2,
     fontSize: 15,
     backgroundColor: 'lightgrey',
   },
+  whiteText: {
+    color: 'white',
```
Comment (ashifkhn) [inline src/screens/ProfileScreen/DetailScreen/ExtensionRequest.tsx:151]: use color variable

File: src/screens/ProfileScreen/DetailsScreen/ProgressDetailScreen.tsx [hunk 14]
```diff
@@ -103,18 +83,13 @@ const styles = StyleSheet.create({
...
-  //   color: 'black',
-  // },
   titleContainer: {
     padding: 10,
     alignItems: 'center',
     marginBottom: 12,
   },
   titleText: {
-    color: '#041187', // You can customize the text color
+    color: '#041187',
```
Comment (ashifkhn) [inline src/screens/ProfileScreen/DetailsScreen/ProgressDetailScreen.tsx:92]: use color variable

File: src/screens/ProfileScreen/TaskScreens/NoteWorthy.tsx [hunk 15]
```diff
@@ -24,61 +24,39 @@ const Note = () => {
...
+    }, [loggedInUserData?.username]),
   );
 
   return (
-    <ScrollView style={{ padding: 10, elevation: 10 }}>
+    <ScrollView style={styles.container}>
       {userContributionData ? (
         <View style={profileScreenStyles.container}>
-          {userContributionData.map((item, index) => (
+          {userContributionData.map((item: any, index) => (
```
Comment (ashifkhn) [inline src/screens/ProfileScreen/TaskScreens/NoteWorthy.tsx]: try not to use any

File: src/screens/ProfileScreen/styles.ts [hunk 16]
```diff
@@ -61,11 +60,11 @@ export const profileScreenStyles = StyleSheet.create({
...
   stickyButtonContainer: {
     position: 'absolute',
-    bottom: 16, // Adjust the value based on your preference
-    left: '20%', // Center the button horizontally
-    transform: [{ translateX: -50 }], // Center the button horizontally
-    width: '80%', // Adjust the width based on your preference
-    zIndex: 2, // Ensure the button is above other elements
+    bottom: 16,
+    left: '20%',
+    transform: [{ translateX: -50 }],
```
Comment (ashifkhn) [inline src/screens/ProfileScreen/styles.ts:65]: dont hard code -50, use scale function

File: src/screens/ProfileScreen/styles.ts [hunk 17]
```diff
@@ -61,11 +60,11 @@ export const profileScreenStyles = StyleSheet.create({
   },
   stickyButtonContainer: {
     position: 'absolute',
-    bottom: 16, // Adjust the value based on your preference
-    left: '20%', // Center the button horizontally
-    transform: [{ translateX: -50 }], // Center the button horizontally
-    width: '80%', // Adjust the width based on your preference
-    zIndex: 2, // Ensure the button is above other elements
+    bottom: 16,
```
Comment (ashifkhn) [inline src/screens/ProfileScreen/styles.ts:63]: dont hard code 16 use scale function, should be responsive

**Review by ashifkhn — CHANGES_REQUESTED** [review]
Some minor fixes

File: src/screens/ProfileScreen/DetailScreen/ActiveTaskDetail.tsx (hunk 11 above)
Comment (Ayushsanjdev) [reply src/screens/ProfileScreen/DetailScreen/ActiveTaskDetail.tsx:93]: for color variable, we need to add a new issue  as we have to update it everywhere with testing.

File: src/screens/ProfileScreen/DetailScreen/ExtensionRequest.tsx (hunk 12 above)
Comment (Ayushsanjdev) [reply src/screens/ProfileScreen/DetailScreen/ExtensionRequest.tsx:93]: for color variable, we need to add a new issue  as we have to update it everywhere with testing.

File: src/screens/ProfileScreen/DetailScreen/ExtensionRequest.tsx (hunk 13 above)
Comment (Ayushsanjdev) [reply src/screens/ProfileScreen/DetailScreen/ExtensionRequest.tsx:151]: for color variable, we need to add a new issue  as we have to update it everywhere with testing.

File: src/screens/ProfileScreen/DetailsScreen/ProgressDetailScreen.tsx (hunk 14 above)
Comment (Ayushsanjdev) [reply src/screens/ProfileScreen/DetailsScreen/ProgressDetailScreen.tsx:92]: for color variable, we need to add a new issue  as we have to update it everywhere with testing.

File: src/screens/ProfileScreen/styles.ts (hunk 17 above)
Comment (Ayushsanjdev) [reply src/screens/ProfileScreen/styles.ts:63]: for this also, we need to add a new issue  as we have to update it everywhere with testing.

File: src/screens/ProfileScreen/styles.ts (hunk 16 above)
Comment (Ayushsanjdev) [reply src/screens/ProfileScreen/styles.ts:65]: for this also, we need to add a new issue as we have to update it everywhere with testing.

**Comment by Ayushsanjdev** [comment]
> Some minor fixes

fixed and replied accordingly

**Review by ashifkhn — APPROVED** [review]

File: src/screens/ProfileScreen/User Data/UserData.tsx [hunk 18]
```diff
@@ -9,17 +9,18 @@ import {
...
-    username,
-  } = userData;
+    twitter_id = '',
+    linkedin_id = '',
+    github_id = '',
+    designation = '',
+    company = '',
+    name = '',
+    username = '',
+  } = userData || {};
```
Comment (prakashchoudhary07) [inline src/screens/ProfileScreen/User Data/UserData.tsx:23]: what is happening here?

Comment (Ayushsanjdev) [reply src/screens/ProfileScreen/User Data/UserData.tsx:23]: Added default values for each property to fix types issue

**Review by ashifkhn — APPROVED** [review]

**Pull request merged**